ocean_lib/web3_internal/web3_overrides/http_provider.py

```diff
@@ -2,14 +2,16 @@
 # Copyright 2021 Ocean Protocol Foundation
 # SPDX-License-Identifier: Apache-2.0
 #
+from typing import Any, Dict
+
 from ocean_lib.web3_internal.web3_overrides.request import make_post_request
 from web3 import HTTPProvider
 
 
 class CustomHTTPProvider(HTTPProvider):
```

Comment:
Add `@enforce_types`

Reply:
I am having trouble enforcing this. I get issues with the inherited init, and the build errors out.

Reply:
I managed to enforce types for the function. Enforcing for the entire class breaks because of HTTPProvider usages and instances from web3 components. Btw. same for the RPCEndpoint type hinting, I reverted that to str since this is what web3 seems to ask for.

Reply:
> I managed to enforce types for the function. Enforcing for the entire class breaks because of HTTPProvider usages and instances from web3 components.

Fair enough.  Thanks for trying.

> Btw. same for the RPCEndpoint type hinting, I reverted that to str since this is what web3 seems to ask for.

I guess that's fine. I like `RPCEndpoint` better because it's more self-documenting. But you're right that it's just a `str` under the hood. https://github.com/ethereum/web3.py/blob/master/web3/types.py#L57